### PR TITLE
Add Qwen3-Next model

### PR DESCRIPTION
**Pull request by lshpku**
### PR types
New features

### PR changes
Models

### Description
新增 Qwen3-Next 模型

### Checklist

| 新增模型步骤 | 验收标准 | 完成情况 |
| --- | --- | :-: |
| 单卡前向精度对齐 | 相比于torch，前向logits diff≤1e-2 | ✅
| 新增模型单测 | PR中添加&跑通 | ✅
| TP前向精度对齐 | 相比于单卡，前向logits diff≤1e-2 | ✅
| 模型生成验证 | 输出正常，指定case，比较greedy输出结果 | ✅
| 混合并行策略TP+SP+PP+EP | SFT训练300步，与单卡或PP训练loss一致 | ✅
| 训练数据流验证 | 数据流拼接符合预期 | ✅
| 模型训练正确性验证 | SFT GSM8K 收敛正常，指标符合预期（>75%） | ✅
| PT -> SFT -> DPO (Full & LoRA) | 全流程串通，收敛正常 | ✅
| CI/CE添加 | 添加缩层模型CI/CE | ✅
| README更新代码能力矩阵 | PR中体现 | 另开PR

### CODE REVIEW

File: paddleformers/transformers/qwen3_next/configuration.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```
+    """
+
+    model_type = "qwen3_next"
+    keys_to_ignore_at_inference = ["past_key_values"]
+
+    base_model_tp_plan = {
```
Comment (lugimzzz) [inline paddleformers/transformers/qwen3_next/configuration.py]: 去掉 base_model_tp_plan、base_model_pp_plan

File: paddleformers/transformers/qwen3_next/configuration.py [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+        self.shared_expert_intermediate_size = shared_expert_intermediate_size
+        self.num_experts_per_tok = num_experts_per_tok
+        self.num_experts = num_experts
+        self.norm_topk_prob = norm_topk_prob
+        self.output_router_logits = output_router_logits
+        self.router_aux_loss_coef = router_aux_loss_coef
+        self.mlp_only_layers = mlp_only_layers
+
+        # for backward compatibility
+        self.sliding_window = None
```
Comment (lugimzzz) [inline paddleformers/transformers/qwen3_next/configuration.py]: 这个三个config的作用？

**Comment by lugimzzz** [comment]
没有适配FlexCheckpoint

File: paddleformers/transformers/qwen3_next/configuration.py (hunk 1 above)
Comment (lshpku) [reply paddleformers/transformers/qwen3_next/configuration.py]: 已改

File: paddleformers/transformers/qwen3_next/configuration.py (hunk 2 above)
Comment (lshpku) [reply paddleformers/transformers/qwen3_next/configuration.py]: 是之前套件不完善的时候写的，现在基类里面已经有定义了，已删除

**Comment by lshpku** [comment]
> 没有适配FlexCheckpoint

默认的 super().sharded_state_dict 方法已经正确，经检查全局 expert_id、pp 的 layer_idx、tp 的 global_offset 都没问题，测试断点重训 loss 也是接续的，因此无需手写 sharded_state_dict
图片 1

**Review by lugimzzz — APPROVED** [review]
LGTM
